### PR TITLE
Linux: Use D-Bus to inhibit screensaver

### PR DESCRIPTION
**Pull request by robxnano**
This is a fix for https://bugs.dolphin-emu.org/issues/10392, which has been a problem for at least 7 years now. I've implemented support for several D-Bus interfaces to inhibit the screensaver on Linux, which covers all the desktop environments that I could find that support it. As the xdg-screensaver method doesn't work at all any more, I've removed it completely.

As this uses Qt's D-Bus module, it requires dolphin-emu-nogui to be linked against libQt6DBus.so as well. This only happens if Dolphin is built with Qt enabled, otherwise it is simply skipped. If this isn't acceptable, I can move it into DolphinQt instead.

### CODE REVIEW

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by Tilka** [comment]
Do you have any insight into why there is such a proliferation of different APIs?

**Comment by randomstuff** [comment]
> As the xdg-screensaver method doesn't work at all any more, I've removed it completely.

It dosn't? I mean, it's supposed to be a (messy) abstraction of what you are doing. It contains snippets such as:

~~~sh
        dbus-send --session \
                  --dest=org.freedesktop.ScreenSaver \
                  --type=method_call \
                  --print-reply \
                  --reply-timeout=2000 \
                  /ScreenSaver \
                  org.freedesktop.ScreenSaver.Inhibit \
                  string:$window_id \
                  string:xdg-screensaver \
                  | grep uint32 | cut -d ' ' -f 5 >| "$screensaver_file.cookie" \
                  2> /dev/null
~~~ 

which is actually what you are doing here: https://github.com/dolphin-emu/dolphin/pull/13076/commits/643fc029443b06f56a4c8ab0a3c82a1df0b89335#diff-729adc8bd1360f1b27c4ff23c72fb382d0d24e9c12f5ad937257894b6187c4e6R94

**Comment by robxnano** [comment]
> Do you have any insight into why there is such a proliferation of different APIs?

@Tilka The [idle-inhibit spec](https://specifications.freedesktop.org/idle-inhibit-spec/latest/) (org.freedesktop.ScreenSaver) was supposed to be a unified effort by Gnome and KDE developers to replace their own (incompatible) APIs, but it hasn't been widely adopted by other desktop environments. The MATE and Xfce ones are just (IMHO unnecessary) forks of the older org.gnome.ScreenSaver interface. Then there's the [XDG Desktop Portal](https://flatpak.github.io/xdg-desktop-portal/docs/index.htmll) (org.freedesktop.portal.Desktop) interface, which was designed to support sandboxing for Flatpak and has a few extra features. But again, the inhibit feature isn't widely supported outside KDE and Gnome yet, and the KDE implementation had a [bug](https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/commit/1549c49f41f544cb9dad54de509f318cc5b6dc38) that prevented it from working until just last month.

In summary, it's just https://xkcd.com/927/. Hopefully in the future the portal API will become widely adopted and we can drop all the others.

**Comment by robxnano** [comment]
@randomstuff The problem with the dbus-send method is that it runs as a separate executable so it can't keep the connection to the D-Bus socket open. That results in the session manager ending the inhibition immediately, as required by the specification:

> Inhibition will stop when the UnInhibit method is called, or the application disconnects from the D-Bus session bus (which usually happens upon exit).

With this new change, the message is sent through the connection opened by Qt for other purposes, and kept open for the life of the application. This ensures the screensaver stays suspended for as long as necessary.

**Comment by theboreddevvv** [comment]
@dolphin-emu-bot rebuild

**Comment by vincent-olivert-riera** [comment]
@robxnano , @Tilka , from which release is this fix included? I am using Dolphin 5.0-17995-g8bad821019 installed from flatpack, using Debian's stable KDE wayland, and my screensaver is not inhibited while playing.

**Comment by jordan-woyak** [comment]
> @robxnano , @Tilka , from which release is this fix included? I am using Dolphin 5.0-17995-g8bad821019 installed from flatpack, using Debian's stable KDE wayland, and my screensaver is not inhibited while playing.

You're over two years out of date.

This PR was merged for version: 2409-136

**Comment by vincent-olivert-riera** [comment]
> > @robxnano , @Tilka , from which release is this fix included? I am using Dolphin 5.0-17995-g8bad821019 installed from flatpack, using Debian's stable KDE wayland, and my screensaver is not inhibited while playing.
> 
> You're over two years out of date.
> 
> This PR was merged for version: 2409-136

Do you mean my version (Dolphin 5.0-17995-g8bad821019) is tool old and does not include the fix? I'm not sure how to compare that version number (2409-136) you gave me.

**Comment by jordan-woyak** [comment]
> Do you mean my version (Dolphin 5.0-17995-g8bad821019) is tool old and does not include the fix? I'm not sure how to compare that version number (2409-136) you gave me.

Yes.
https://dolphin-emu.org/download/dev/master/5.0-17995/
https://dolphin-emu.org/download/dev/master/2409-136/
